components/AudioRecorder/index.tsx

```diff
@@ -59,7 +88,72 @@ export const AudioRecorderComponent: React.FC = () => {
   const [consultationName, setConsultationName] = useState('');
   const [isUploading, setIsUploading] = useState(false);
   const [notes, setNotes] = useState(''); // State for textarea
+  const modalRef = useRef<HTMLDivElement>(null);
+
+  // New state for sidebar
+  const [isSidebarExpanded, setIsSidebarExpanded] = useState(false);
+  const [sidebarContent, setSidebarContent] = useState<
+    'notes' | 'checklist' | null
+  >(null);
+  const sidebarRef = useRef<HTMLDivElement>(null);
+
+  //! Function to toggle sidebar
+  const toggleSidebar = (content: 'notes' | 'checklist') => {
+    if (isSidebarExpanded && sidebarContent === content) {
+      closeSidebar();
+    } else {
+      setIsSidebarExpanded(true);
+      setSidebarContent(content);
+    }
+  };
+
+  // !Function to close sidebar
+  const closeSidebar = () => {
+    setIsSidebarExpanded(false);
+    setSidebarContent(null);
+  };
+
+  //! Side states for responsive Design
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const { width } = useWindowSize();
+  const isMobile = width < 768; // Define the breakpoint for mobile screens
+
+  const showModal = (content: 'notes' | 'checklist') => {
+    setSidebarContent(content);
+    setIsModalVisible(true);
+  };
+
+  const handleModalCancel = () => {
+    setIsModalVisible(false);
+    setSidebarContent(null);
+  };
 
+  // !Effect to handle clicking outside the sidebar
+
+  useEffect(() => {
```

Comment:
Try to make this code cleaner for readabillity:


```suggestion

useEffect(() => {
  const handleClickOutside = (event: MouseEvent) => {
    if (!isSidebarExpanded) return;

    const clickedOutsideSidebar =
      sidebarRef.current && !sidebarRef.current.contains(event.target as Node);

    const clickedOutsideModal =
      isModalVisible &&
      modalRef.current &&
      !modalRef.current.contains(event.target as Node);

    if (clickedOutsideSidebar && !clickedOutsideModal) {
      closeSidebar();
    }
  };

  // Add event listener for mouse clicks
  document.addEventListener('mousedown', handleClickOutside);

  // Cleanup event listener
  return () => {
    document.removeEventListener('mousedown', handleClickOutside);
  };
}, [isSidebarExpanded, isModalVisible, sidebarRef, modalRef, closeSidebar]);
```